--- a/internal/descriptor/apiconfig/BUILD.bazel
+++ b/internal/descriptor/apiconfig/BUILD.bazel
@@ -1,29 +1,17 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
-load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
-load("@rules_proto//proto:defs.bzl", "proto_library")

 package(default_visibility = ["//visibility:public"])

-proto_library(
-    name = "apiconfig_proto",
-    srcs = [
-        "apiconfig.proto",
-    ],
-    deps = ["@googleapis//google/api:http_proto"],
-)
-
-go_proto_library(
-    name = "apiconfig_go_proto",
-    compilers = ["//:go_apiv2"],
-    importpath = "github.com/grpc-ecosystem/grpc-gateway/v2/internal/descriptor/apiconfig",
-    proto = ":apiconfig_proto",
-    deps = ["@org_golang_google_genproto_googleapis_api//annotations"],
-)
-
+# Use the pre-generated Go file to avoid @googleapis dependency.
 go_library(
     name = "apiconfig",
-    embed = [":apiconfig_go_proto"],
+    srcs = ["apiconfig.pb.go"],
     importpath = "github.com/grpc-ecosystem/grpc-gateway/v2/internal/descriptor/apiconfig",
+    deps = [
+        "@org_golang_google_genproto_googleapis_api//annotations",
+        "@org_golang_google_protobuf//reflect/protoreflect",
+        "@org_golang_google_protobuf//runtime/protoimpl",
+    ],
 )

 alias(
