--- server/pkg/util/secrets_test.go
+++ server/pkg/util/secrets_test.go
@@ -336,6 +336,20 @@
		_, err := util.ResolveSecretMap(context.Background(), secretMap, nil)
		assert.Error(t, err)
	})
+
+	t.Run("ResolveSecret_RegexCache", func(t *testing.T) {
+		pt := &configv1.SecretValue{}
+		pt.SetPlainText("secret_pt")
+		pt.SetValidationRegex("sec.*")
+
+		result1, err1 := util.ResolveSecret(context.Background(), pt)
+		assert.NoError(t, err1)
+		assert.Equal(t, "secret_pt", result1)
+
+		result2, err2 := util.ResolveSecret(context.Background(), pt)
+		assert.NoError(t, err2)
+		assert.Equal(t, "secret_pt", result2)
+	})
 }

 func TestResolveSecret_Vault(t *testing.T) {
